scripts/jenkins/cloud/ansible/roles/heat-generator/templates/heat-template.yaml

```diff
@@ -71,8 +72,22 @@ resources:
       network_id: { get_resource: {{ name }}_network }
 {%   if network.cidr is defined %}
       cidr: "{{ network.cidr }}"
+{%       if ( network.cidr | ipv6 ) %}
+      ip_version: 6
+{%          if network.is_conf or network.is_mgmt %}
+      ipv6_ra_mode: dhcpv6-stateful
+      ipv6_address_mode: dhcpv6-stateful
+{%          endif %}
+{%       else %}
+      ip_version: 4
+{%       endif %}
 {%   elif network.allocation_pools is not defined %}
       cidr: { get_param: default_subnet_cidr }
+      ip_version: {{ ipv6 | ternary(6, 4) }}
+{%       if network.is_conf or network.is_mgmt %}
```

Comment:
same here: the `default_subnet_cidr` parameter is only used because the input model doesn't specify a CIDR for this network. It's just a dummy value. This would never happen with a lifecycle management network (`.is_conf=true`) or the service management network (`.is_mgmt=true`), so you can safely remove this whole if block, it will never be called.